### PR TITLE
Toast For all!

### PR DESCRIPTION
**Pull request by FJMentz**
Installed the ngx-toast package to add nice toasts (notifications) as well as added basic toasts where I could and also added some success messages for when an action's result isn't clear to the user.

Notes can be found [here](https://www.npmjs.com/package/ngx-toastr).

To overcome the injection errors the next is needed in the spec file:
`...
import { ToastrService, ToastrModule } from 'ngx-toastr';
`
and in beforeEach:
`
...
imports: [..., ToastrModule.forRoot(), ...],
providers: [
...,
ToastrService,
 ...
],
...
`

Please feel free to review and leave comments. :)

### CODE REVIEW

**Review by egeldenhuys — CHANGES_REQUESTED** [review]
Toasts look nice. Now I no longer feel like I have to close every alert manually. The toasts also persist across page navigation which removed the need for the bloated alert logic.

Some changes to be made:
- [x] Remove existing alerts 
- [x] Add error reason to toasts. For example instead of `Error deploying stack` we can say `Error deploying stack: Top-level object must be a mapping` => api
- [x] Pass error message for Create User
- [x] Remove empty alert on Stack page

Screenshots:
#### User Create `stub` message
screen shot 2018-09-19 at 16 08 47

##### Stack create random empty alert
screen shot 2018-09-19 at 16 06 19

**Comment by FJMentz** [comment]
@egeldenhuys , I think the last commit addresses the Stack's empty alert and old alerts? Please advise if this is not the case.

File: src/app/pages/stacks/stacks-create/stacks-create.component.ts [hunk 1]
```diff
@@ -42,37 +37,15 @@ export class StacksCreateComponent implements OnInit {
...
-    this.genericError = false;
-    this.submitted = true;
-
     this.stackService
       .deployStack(this.stackModel.stackName, btoa(this.stackModel.stackFile))
       .subscribe(
         (result: StackError) => {
-          this.clearAlerts();
-          this.submitted = false;
-          this.router.navigate([
```
Comment (egeldenhuys) [inline src/app/pages/stacks/stacks-create/stacks-create.component.ts:60]: We still need to navigate to the Stack List view after successfully deploying a stack, but we can remove `{ createdStack: this.stackModel.stackName },`

File: src/app/pages/stacks/stacks-create/stacks-create.component.ts [hunk 2]
```diff
@@ -42,37 +37,15 @@ export class StacksCreateComponent implements OnInit {
...
-          this.clearAlerts();
-          if (err.code === StackErrorCode.ERR_STACK_NAME_TAKEN) {
-            this.alreadyExists = true;
-            this.badUser = this.stackModel.stackName;
-          } else {
-            this.warning = true;
-            this.warningMessage = err.message;
-          }
-          this.submitted = false;
+          this.toastr.error(err.message, 'Error while deploying stack');
```
Comment (egeldenhuys) [inline src/app/pages/stacks/stacks-create/stacks-create.component.ts:49]: I will open a PR now to provide proper error reporting for the stack pages

**Review by egeldenhuys — CHANGES_REQUESTED** [review]
TODO:
- [x] Remove alert from `/users`
- [x] Add toast for deleting user
- [x] Remove alert from `/volumes`
- [x] Remove alert from `/volumes/create`

File: src/app/pages/stacks/stacks-create/stacks-create.component.ts (hunk 1 above)
Comment (FJMentz) [reply src/app/pages/stacks/stacks-create/stacks-create.component.ts:60]: Does one want to go back directly after deploying a stack?

Consider the situation where you want to deploy 3 stacks. This means that you have to go to the page every time to deploy your new stack?

Or do we assume a person won't deploy more than one stack at a time that often?

File: src/app/pages/stacks/stacks-create/stacks-create.component.ts (hunk 2 above)
Comment (FJMentz) [reply src/app/pages/stacks/stacks-create/stacks-create.component.ts:49]: Ah yess I left it so to be ready for the update :D

**Review by FJMentz — COMMENTED** [review]
Thank you @CDuPlooy !

**Comment by FJMentz** [comment]
This SHOULD be the last addressing of all the requests above. Please review. :) @egeldenhuys @CDuPlooy

**Review by CDuPlooy — APPROVED** [review]
Looks great

**Review by egeldenhuys — APPROVED** [review]
Looks good

**Pull request merged**